### PR TITLE
Fix version handling for packages.drupal.org

### PR DESCRIPTION
**Pull request by bdragon28**
This should continue to behave as before for packages in the drupal/ namespace that are NOT coming from packages.drupal.org.

### CODE REVIEW

**Comment by fabianx-ai** [comment]
**RTBM** - Looks great to me!

File: src/Drupal/Composer/DrupalInstallerPlugin.php [hunk 1]
```diff
@@ -603,29 +603,56 @@ protected function getPackageName(PackageEvent $event) {
...
+        // Detect endpoint.
+        $platform = FALSE;
+        $repo = $package->getRepository();
+        if (method_exists($repo, 'getRepoConfig')) {
+            $repo_config = $repo->getRepoConfig();
+            if (isset($repo_config['url'])) {
+                // The new packages.drupal.org separates the platform from the version entirely.
+                $repourl = parse_url($repo_config['url']);
+                if ($repourl['host'] == 'packages.drupal.org') {
+                    $platform = substr($repourl['path'], 1);
```
Comment (douggreen) [inline src/Drupal/Composer/DrupalInstallerPlugin.php:615]: Why is this substr(..., 1).

Comment (bdragon28) [reply src/Drupal/Composer/DrupalInstallerPlugin.php:615]: just wanted to grab the stuff after the slash.

Comment (bdragon28) [reply src/Drupal/Composer/DrupalInstallerPlugin.php:615]: I was drawing a blank on a less stupid method of grabbing the platform version out of the url.

**Pull request merged**